projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -582,7 +579,7 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
      * @hidden @internal
      */
     public isDetailRecord(record) {
-        return record && record.detailsData !== undefined;
```

Comment:
I remember putting this check on a purpose when record was coming as `undefined`. I will double check if we need it.